### PR TITLE
Updating extension template readme

### PR DESCRIPTION
**Pull request by timgdavies**
I've updated the README so that:

* We warn against removing properties, or overwriting codelists;
* It documents the '+codelist.csv' syntax for adding values to codelists;
* The examples don't show overwriting closed codelists;
* The documentation is otherwise aligned with 1.1.x

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,24 +100,25 @@ The actual description in the extension is much better:
 
 As in the [core standard repository](https://github.com/open-contracting/standard), the extension template also includes a codelists folder to store extension-specific codelists.
 
-Codelists are CSV files with camel case names , e.g. _contractStatus.csv_. Be aware that a codelist in your extension using the same name of an existing codelist in the standard repository will override the existing codelist.
+Codelists are CSV files with camel case names , e.g. _implementationStatus.csv_. Be aware that a codelist in your extension using the same name of an existing codelist in the standard repository will override the existing codelist.
+
+Prefixing a codelist name with _+_ (e.g. _+milestoneStatus.csv_) indicates that contents of the codelist should be appended to an existing codelist of the same name.
```
Comment (timgdavies) [inline README.md]: @Bjwebb @edugomez Re: https://github.com/OpenDataServices/cove/issues/528#issuecomment-346628142 can you confirm whether CoVE is yet following this pattern?

File: README.md [hunk 2]
```diff
@@ -2,21 +2,25 @@
 
 This repository has two purposes:
 
-* To give a template of an extension that people can base their own on.
-* To give a place for comment and discussion on how extensions should work.
+* To provide a template of an extension that people can base their own on.
+* To provide a place for comment and discussion on how extensions should work.
```
Comment (jpmckinney) [inline README.md]: Hmm, https://github.com/open-contracting/ocds-extensions/issues may be the better place for that now.

File: README.md [hunk 3]
```diff
@@ -2,21 +2,25 @@
...
+
+[OCDS extensions](http://standard.open-contracting.org/latest/en/extensions/) can be used to declare additional fields, objects and sections that will be included in an OCDS file, but which are not in the core schema, and to add or extend codelists. When any OCDS publication cites an extension in its' package metadata, it should be validated against the schema with that extension applied. 
 
 ## Getting started
 
 To create an extension based on this template, you should:
 
 1. Download [a zip file version of the template](https://github.com/open-contracting/standard_extension_template/archive/master.zip)
-1. Extract it, and initialise it as a git repository (`git init`)
+1. Extract it, and initialise it as a git repository (`git init`)*
```
Comment (jpmckinney) [inline README.md]: The bare asterisk may create italicization in Markdown. Maybe use dagger † instead.

File: README.md [hunk 4]
```diff
@@ -50,7 +55,7 @@ Extensions must include at least one of those files. In most cases, the extensio
 
 This file is required. It provides information about the extension to be used by any OCDS automated tool.
 
-This repository contains an [example extension.json](https://github.com/open-contracting/standard_extension_template/blob/master/extension.json). [extension-schema.json](https://github.com/open-contracting/standard-maintenance-scripts/blob/master/schema/extension-schema.json) describes its format.
+This repository contains an [example extension.json](https://github.com/open-contracting/standard_extension_template/blob/master/extension.json) that describes its format.
```
Comment (jpmckinney) [inline README.md]: The example doesn't describe its own format – only the schema offers a description.

File: README.md [hunk 5]
```diff
@@ -95,24 +100,25 @@ The actual description in the extension is much better:
 
 As in the [core standard repository](https://github.com/open-contracting/standard), the extension template also includes a codelists folder to store extension-specific codelists.
 
-Codelists are CSV files with camel case names , e.g. _contractStatus.csv_. Be aware that a codelist in your extension using the same name of an existing codelist in the standard repository will override the existing codelist.
+Codelists are CSV files with camel case names , e.g. _implementationStatus.csv_. Be aware that a codelist in your extension using the same name of an existing codelist in the standard repository will override the existing codelist.
```
Comment (jpmckinney) [inline README.md]: Let's remove the space before the comma and "same name of" -> "same name as". Can add ", which is not recommended" at end of paragraph.

File: README.md [hunk 6]
```diff
@@ -95,24 +100,25 @@ The actual description in the extension is much better:
...
+
+Prefixing a codelist name with _+_ (e.g. _+milestoneStatus.csv_) indicates that contents of the codelist should be appended to an existing codelist of the same name. 
 
 ## How the extensions work
 
-In the core repository as of OCDS version 1.0.1 there are the following 4 schema files.
+In the core repository as of OCDS version 1.1.1 there are the following 3 schema files.
```
Comment (jpmckinney) [inline README.md]: I assume core refers to `standard` in which case it's true that there are four. Maybe rephrase as, "As of OCDS 1.1.1, the 3 schema files from the [core standard](https://github.com/open-contracting/standard/tree/1.1-dev/standard/schema) that extensions can extend are:"

File: README.md [hunk 7]
```diff
@@ -95,24 +100,25 @@ The actual description in the extension is much better:
...
 
 * release-schema.json
 * record-package-schema.json
 * release-package-schema.json
-* versioned-release-validation-schema.json
 
 In the extension some or all of these files can be used to do a JSON Merge Patch of the corresponding file.
 
 A JSON merge patch is described by this [rfc](https://tools.ietf.org/html/rfc7386).
```
Comment (jpmckinney) [inline README.md]: Let's take opportunity to properly case JSON Merge Patch and RFC.

File: README.md [hunk 8]
```diff
@@ -95,24 +100,25 @@ The actual description in the extension is much better:
...
 
 The patches are very simple. They just copy the same structure from the core schema and allow you add your extra fields or update existing fields just in the places the extension wants to change the schema.
 
-Here are some simple examples of how this works. They are illistrative, any changes to existing fields should respect the [Conformance documentation](http://standard.open-contracting.org/latest/en/schema/conformance_and_extensions/).
+Here are some simple examples of how this works. They are illustrative, any changes to existing fields should respect the [Conformance documentation](http://standard.open-contracting.org/latest/en/schema/conformance_and_extensions/).
```
Comment (jpmckinney) [inline README.md]: Replace comma with semi-colon before "any"

File: README.md [hunk 9]
```diff
@@ -158,6 +164,8 @@ To update the description of the planning phase.
 }
 ```
 
+Note: Core or community extensions MAY add clarifying notes to a description, but MUST NOT change the definition of a field, as this would fail against the [conformance criteria](http://standard.open-contracting.org/latest/en/schema/conformance_and_extensions/).
```
Comment (jpmckinney) [inline README.md]: Hmm, but multiple extensions might each change a field's description, and only the last applied extension's change will survive. I think such clarifications belong in documentation.

File: README.md [hunk 10]
```diff
@@ -171,6 +179,8 @@ Remove whole planning section. Putting in null will remove the key associated wi
 }
 ```
 
+Note: Core or community extensions MUST NOT remove any existing sections or fields. This feature should only be used by local extensions, or in rare cases by OCDS profiles.
```
Comment (jpmckinney) [inline README.md]: Should we even be including this example?

File: README.md (hunk 1 above)
Comment (Bjwebb) [reply README.md]: Sorry, thought this had a response. No, CoVE isn't doing this yet, here's an issue for it https://github.com/OpenDataServices/cove/issues/914.

@kindly Actually, are we checking codelists in extensions at all atm?

**Comment by jpmckinney** [comment]
Forgot about this PR. @timgdavies Can you fix my few comments and resolve conflicts?

**Comment by jpmckinney** [comment]
I'll fix conflicts and merge. However, I removed these two problematic examples. No extensions presently change `title` or `description` or remove fields (except maybe OCDS for PPPs, which is somewhat special).

```markdown
To update the description of the planning phase.

```json
{
  "properties": {
    "planning": {
      "description": "Information from the planning phase of the contracting process. This may include detailed budgets broken down by year (supported by the budgets extension).”
    }
  }
}
```

Note: Core or community extensions MAY add clarifying notes to a description, but MUST NOT change the definition of a field, as this would fail against the [conformance criteria](http://standard.open-contracting.org/latest/en/schema/conformance_and_extensions/).

Remove whole planning section. Putting in null will remove the key associated with it.

```json
{
  "properties": {
    "planning": null
  },
  "definitions": {
    "Planning": null
  }
}
```

Note: Core or community extensions MUST NOT remove any existing sections or fields. This feature should only be used by local extensions, or in rare cases by OCDS profiles.
```
